src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Mets/MetsRepository.cs

```diff
@@ -95,28 +105,63 @@ public class MetsRepository : IMetsRepository
+                .Where(el => IsLabelledDirectory(el, "submissionDocumentation")).ToList();
+            
+            if (metadataDirectories.Count == 0)
             {
-                metadataDirectory = objectsChildren[^2..].SingleOrDefault(el =>
-                    el.Attribute(TypeAttribute)?.Value == Directory && el.Attribute(LabelAttribute)?.Value == "metadata");
-                submissionDocumentationDirectory = objectsChildren[^2..].SingleOrDefault(el =>
-                    el.Attribute(TypeAttribute)?.Value == Directory && el.Attribute(LabelAttribute)?.Value == "submissionDocumentation");
+                throw new NotSupportedException("Could not find directory labelled 'metadata' in objects directory");
             }
-            if (metadataDirectory == null || submissionDocumentationDirectory == null)
+            if (submissionDocumentationDirectories.Count == 0)
             {
-                throw new NotSupportedException("Objects directory does not have metadata and submissionDocumentation as last two entries");
+                throw new NotSupportedException("Could not find directory labelled 'submissionDocumentation' in objects directory");
             }
-
+            if (metadataDirectories.Count > 1)
+            {
+                throw new NotSupportedException("More than one directory labelled 'metadata' in objects directory");
+            }
+            if (submissionDocumentationDirectories.Count > 1)
+            {
+                throw new NotSupportedException("More than one directory labelled 'submissionDocumentation' in objects directory");
+            }
+            
+            if (ddsOptions.ArchivematicaStrictDirectoryOrder)
+            {
+                // These should be the last two, but we won't mind the order
+                if (objectsChildren.Length >= 2)
+                {
+                    metadataDirectory = objectsChildren[^2..]
+                        .SingleOrDefault(el => IsLabelledDirectory(el, "metadata"));
+                    submissionDocumentationDirectory = objectsChildren[^2..]
+                        .SingleOrDefault(el => IsLabelledDirectory(el, "submissionDocumentation"));
+                }
+                if (metadataDirectory == null || submissionDocumentationDirectory == null)
+                {
+                    throw new NotSupportedException("Objects directory does not have metadata and submissionDocumentation as last two entries");
+                }
+                // We can now ignore the last two.
+                digitalContent = objectsChildren[..^2];
+            }
+            else
+            {
+                metadataDirectory = metadataDirectories.Single();
+                submissionDocumentationDirectory = submissionDocumentationDirectories.Single();
+                digitalContent = objectsChildren.Where(el =>
+                    !(IsLabelledDirectory(el, "metadata") || IsLabelledDirectory(el, "submissionDocumentation")))
```

Comment:
Is it worth making consts for `"metadata"` and `"submissionDocumentation"` as they're used in 6 times?

Reply:
yeah will do